Source/RapyutaSimulationPlugins/Public/Robots/RRRobotStructs.h

```diff
@@ -618,20 +651,65 @@ struct RAPYUTASIMULATIONPLUGINS_API FRRSensorLidarInfo
     double NoiseMean = 0;
     UPROPERTY(EditAnywhere)
     double NoiseStdDev = 0;
+
+    virtual void PrintSelf() const override
+    {
+        Super::PrintSelf();
+        UE_LOG_WITH_INFO(
+            LogTemp, Display, TEXT("- LidarType: %s"), *URRTypeUtils::GetEnumValueAsString(TEXT("ERRLidarSensorType"), LidarType));
+
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("- Scan horizontal"));
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ NHorSamplesPerScan: %d"), NHorSamplesPerScan);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ HMinAngle: %f"), HMinAngle);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ HMaxAngle: %f"), HMaxAngle);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ HResolution: %f"), HResolution);
+
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("- Scan vertical"));
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ NVerSamplesPerScan: %d"), NVerSamplesPerScan);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ VMinAngle: %f"), VMinAngle);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ VMaxAngle: %f"), VMaxAngle);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ VResolution: %f"), VResolution);
+
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("- Range"));
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ MinRange: %f"), MinRange);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ MinRange: %f"), MaxRange);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ Resolution: %f"), RangeResolution);
+
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("- Noise"));
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ Type: %s"), *NoiseTypeName);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ Mean: %f"), NoiseMean);
+        UE_LOG_WITH_INFO(LogTemp, Display, TEXT("\t+ StdDev: %f"), NoiseStdDev);
+    }
 };
 
 USTRUCT(BlueprintType)
 struct RAPYUTASIMULATIONPLUGINS_API FRRSensorProperty
 {
     GENERATED_BODY()
+
     UPROPERTY(EditAnywhere)
     FString LinkName;
     UPROPERTY(EditAnywhere)
     FString SensorName;
     UPROPERTY(EditAnywhere)
     ERRSensorType SensorType = ERRSensorType::NONE;
+
+    //! TODO: To avoid extra memory cost, this should be actually a TUniquePtr, which is only instantiated upon SensorType as LIDAR.
+    //! However, it will become not visible/editable in Blueprint like BP data table
     UPROPERTY(EditAnywhere)
     FRRSensorLidarInfo LidarInfo;
```

Comment:
What is the future plan to support general sensor inforation? Can TUniquePtr support multiple sensor type as your comment?

Reply:
It could be like:
```
TUniquePtr<FRRSensorBaseInfo> LidarInfo;
TUniquePtr<FRRSensorBaseInfo> CameraInfo;
TUniquePtr<FRRSensorBaseInfo> IMUInfo;
etc.
```
then `LidarInfo = MakeUnique<FRRSensorLidarInfo>();` & so on
but I'm afraid it could never be supported as long as it is UPROPERTY() to be editable in BP

Reply:
It could be like:
```
TUniquePtr<FRRSensorBaseInfo> LidarInfo;
TUniquePtr<FRRSensorBaseInfo> CameraInfo;
TUniquePtr<FRRSensorBaseInfo> IMUInfo;
etc.
```
then `if (ERRSensorType::LIDAR == SensorType) LidarInfo = MakeUnique<FRRSensorLidarInfo>();` & so on
but I'm afraid it could never be supported as long as it is UPROPERTY() to be editable in BP